### ISSUE TITLE
Random Http2NettyTest.deadlineExceeded() failure.

### ISSUE THREAD

**Issue by madongfly** [grpc/grpc-java#935]
Saw this failure at https://travis-ci.org/grpc/grpc-java/builds/78476097, may wroth taking a look.

```
io.grpc.testing.integration.Http2NettyTest > deadlineExceeded FAILED
    java.lang.AssertionError: expected:<Status{code=DEADLINE_EXCEEDED, description=null, cause=null}> but was:<Status{code=UNKNOWN, description=null, cause=io.netty.util.IllegalReferenceCountException: refCnt: 0}>
```

**Comment by nmittler** [comment]
Any chance this is related to #926?

**Comment by ejona86** [comment]
Happened in https://travis-ci.org/grpc/grpc-java/builds/79707682 as well:
```
io.grpc.testing.integration.Http2NettyLocalChannelTest > deadlineExceeded FAILED
    java.lang.AssertionError: expected:<Status{code=DEADLINE_EXCEEDED, description=null, cause=null}> but was:<Status{code=UNKNOWN, description=null, cause=java.lang.IllegalStateException: Cannot transition phase from STATUS to MESSAGE}>
```

**Comment by ejona86** [comment]
So my last comment is probably #765. I've not seen this re-appear.

**Comment by ejona86** [comment]
This is caused by deadline-caused cancel() failing future messages that are written. We either need to not change the outboundPhase for cancel() or remove outboundPhase() outright, since ClientCallImpl checks state.

https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/internal/AbstractClientStream.java#L296

**Comment by ejona86** [comment]
(bug change was because I thought this was #765)

**Comment by nmittler** [comment]
This should be fixed now (via #1071).  We can re-open if it's still an issue.
